src/csharp/Grpc.Core/Internal/NativeExtension.cs

```diff
@@ -114,8 +114,14 @@ private static NativeMethods LoadNativeMethods()
         /// </summary>
         private static NativeMethods LoadNativeMethodsUnity()
         {
-            // TODO(jtattermusch): use NativeMethods.DllImportsFromStaticLib for iOS.
-            return new NativeMethods(new NativeMethods.DllImportsFromSharedLib());
+            switch (PlatformApis.GetUnityRuntimePlatform())
+            {
+                case "IPhonePlayer":
```

Comment:
I'm _pretty sure_ `tvOS` and `WebGLPlayer` will both want `__Internal` as well, FWIW, but I'm guessing that's something for a later PR?